ports/stm32/sdcard.c

```diff
@@ -185,6 +184,12 @@ void HAL_SD_MspInit(SD_HandleTypeDef *hsd) {
     // GPIO have already been initialised by sdcard_init
 }
 
+void sdcard_allocate_handle() {
+    // create the sd_handle
+    static SD_HandleTypeDef _sd_handle = {0};
```

Comment:
I'm pretty sure this will statically allocate the data structure on the BSS regardless of this function being called or not.  Otherwise, how does it know where to allocate it at runtime?

Reply:
hehe... yeah of course it is. 
Now I'm thinking, the first time I saw this technique was for a c++ static class construction that was somewhat cpu intensive, so it was deferring that as the main goal rather than ram.